subsys/bluetooth/shell/bt.c

```diff
@@ -697,6 +697,56 @@ static int cmd_name(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+static int hex4str_to_uint16(const char * str, uint16_t * output)
+{
+	/* Ignore 0x at start. */
+	if (strlen(str) == 6) {
+		if (str[0] == '0' && str[1] == 'x') {
+			str = &str[2];
+		}
+	}
+
+	/* Accept exactly 4 hex digits. */
+	if (strlen(str) == 4) {
+		for (size_t i = 0; i < strlen(str); i++) {
+			if (!isxdigit(str[i])) {
+				return -EINVAL;
+			}
+		}
+
+		*output = (uint16_t) strtol(str, NULL, 16);
```

Comment:
Not sure if it is specifically stated in the coding guidelines, but typically we don't have space between the value and the type cast. 
```suggestion
		*output = (uint16_t)strtol(str, NULL, 16);
```